src/main/java/com/solidbrain/russads/AutomaticTaskCreator.java

```diff
@@ -0,0 +1,131 @@
+import com.getbase.sync.Meta;
+import com.getbase.sync.Sync;
+import com.solidbrain.russads.configuration.ConfigurationHelper;
+import com.solidbrain.russads.configuration.ConfigurationPropertyRepository;
+import com.solidbrain.russads.configuration.model.ConfigurationProperty;
+
+@Configuration
+@EnableScheduling
+public class AutomaticTaskCreator {
+
+	@Autowired
+	ConfigurationHelper configurationHelper;
+
+	@Autowired
+	ConfigurationPropertyRepository dao;
+
+	@Value("${BASECRM_ACCESS_TOKEN}")
+	private String BASECRM_ACCESS_TOKEN;
+
+	@Value("${BASECRM_DEVICE_UUID}")
+	private String BASECRM_DEVICE_UUID;
+
+	private final DateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+
+	private Client client;
+
+	private List<ConfigurationProperty> salesRepIdsWithAutomaticTasks;
+
+	@PostConstruct
+	private void initClient() {
+		client = new Client(new com.getbase.Configuration.Builder()
+				.accessToken(BASECRM_ACCESS_TOKEN)
+				.verbose()
+				.build());
+		configurationHelper.fakeFlayway();
+		salesRepIdsWithAutomaticTasks = dao.findByKey(AUTO_SALES_REP_KEY);
+
+	}
+
+	@Scheduled(fixedDelay = 5000)
+	public void checkForCompanyUpdates() {
+		new Sync(client, BASECRM_DEVICE_UUID)
+				.subscribe(Contact.class, this::processContact)
+				.subscribe(Account.class, (meta, account) -> true)
+				.subscribe(Address.class, (meta, address) -> true)
+				.subscribe(AssociatedContact.class, (meta, associatedContact) -> true)
+				.subscribe(Deal.class, (meta, deal) -> true)
+				.subscribe(LossReason.class, (meta, lossReason) -> true)
+				.subscribe(Note.class, (meta, note) -> true)
+				.subscribe(Pipeline.class, (meta, pipeline) -> true)
+				.subscribe(Source.class, (meta, source) -> true)
+				.subscribe(Stage.class, (meta, stage) -> true)
+				.subscribe(Tag.class, (meta, tag) -> true)
+				.subscribe(Task.class, (meta, task) -> true)
+				.subscribe(User.class, (meta, user) -> true)
+				.subscribe(Lead.class, (meta, lead) -> true)
+				.fetch();
+	}
+
+	private boolean processContact(Meta meta, Contact contact) {
+		if (isCreationEvent(meta) &&
+				isContactACompany(contact) &&
+				isAutomaticSalesRepId(contact.getOwnerId()) &&
+				findDealsByContact(contact.getContactId()).isEmpty()) {
+			createDeal(contact);
+		}
+		return true;
+	}
+
+	private void createDeal(Contact contact) {
+		Deal deal = new Deal();
+		deal.setContactId(contact.getId());
+		deal.setOwnerId(contact.getOwnerId());
+		deal.setName(createDealName(contact.getName()));
+		client.deals().create(deal);
+	}
+
+	private String createDealName(String contactName) {
+		return contactName + " " + dateFormat.format(new Date());
+	}
+
+	private List<Deal> findDealsByContact(Long contactId) {
+		return client.deals().list(new DealsService.SearchCriteria().contactId(contactId));
+	}
+
+	private boolean isAutomaticSalesRepId(Long userId) {
+		return salesRepIdsWithAutomaticTasks.stream().filter(r -> r.getValue().equals(userId.toString())).count() > 0;

```

Comment:
Instead of `filter()` and `count() > 0` following method of `Stream` might be used: `anyMatch()`